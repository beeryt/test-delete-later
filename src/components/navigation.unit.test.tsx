import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import * as gatsby from 'gatsby'
const useStaticQuery = jest.spyOn(gatsby, 'useStaticQuery')

import Navigation from './navigation'

describe('Navigation', () => {
  const defaultMetadata = {
    title: 'Default Title',
    description: 'Default Description',
    siteUrl: 'https://example.com'
  }

  beforeAll(() => {
    useStaticQuery.mockImplementation(() => {
      return { site: { siteMetadata: defaultMetadata } }
    })
  })

  beforeEach(() => jest.clearAllMocks())

  afterAll(() => useStaticQuery.mockReset())

  it('renders correctly', async () => {
    render(<Navigation />)
    expect(document.body).toMatchSnapshot()
  })

  it('handles user click', async () => {
    const { asFragment, getByRole } = render(<Navigation />)
    const firstRender = asFragment()
    fireEvent.click(getByRole('button', { name: 'menu' }))
    expect(firstRender).toMatchDiffSnapshot(asFragment())
    fireEvent.click(getByRole('button', { name: 'menu' }))
    expect(firstRender).toMatchDiffSnapshot(asFragment())
  })
})
