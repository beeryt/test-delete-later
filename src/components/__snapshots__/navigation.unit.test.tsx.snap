// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation handles user click 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -15,11 +15,11 @@
            Default Title
          </a>
          <div
            aria-expanded=\\"false\\"
            aria-label=\\"menu\\"
-           class=\\"navbar-burger \\"
+           class=\\"navbar-burger is-active\\"
            role=\\"button\\"
          >
            <span
              aria-hidden=\\"true\\"
            />
@@ -30,11 +30,11 @@
              aria-hidden=\\"true\\"
            />
          </div>
        </div>
        <div
-         class=\\"navbar-menu \\"
+         class=\\"navbar-menu is-active\\"
        >
          <div
            class=\\"navbar-end\\"
          />
        </div>"
`;

exports[`Navigation handles user click 2`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`Navigation renders correctly 1`] = `
<body>
  <div>
    <header>
      <nav
        aria-label="main navigation"
        class="navbar is-primary"
        role="navigation"
      >
        <div
          class="navbar-brand"
        >
          <a
            class="navbar-item is-size-3 has-text-dark"
            href="/"
          >
            Default Title
          </a>
          <div
            aria-expanded="false"
            aria-label="menu"
            class="navbar-burger "
            role="button"
          >
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
          </div>
        </div>
        <div
          class="navbar-menu "
        >
          <div
            class="navbar-end"
          />
        </div>
      </nav>
    </header>
  </div>
</body>
`;
